nameserver/dns.go

```diff
@@ -51,22 +57,19 @@ func filter(ss []string, s string) []string {
 
 func NewDNSServer(ns *Nameserver, domain, address, effectiveAddress string, ttl uint32, clientTimeout time.Duration) (*DNSServer, error) {
 	s := &DNSServer{
-		ns:        ns,
-		domain:    dns.Fqdn(domain),
-		ttl:       ttl,
-		address:   address,
-		tcpClient: &dns.Client{Net: "tcp", ReadTimeout: clientTimeout},
-		udpClient: &dns.Client{Net: "udp", ReadTimeout: clientTimeout, UDPSize: udpBuffSize},
-	}
-	var err error
-	if s.upstream, err = dns.ClientConfigFromFile(etcResolvConf); err != nil {
-		return nil, err
-	}
-	if s.upstream != nil {
-		s.upstream.Servers = filter(s.upstream.Servers, effectiveAddress)
+		ns:               ns,
+		domain:           dns.Fqdn(domain),
+		ttl:              ttl,
+		address:          address,
+		effectiveAddress: effectiveAddress,
+		upstream:         &dns.ClientConfig{},
+		tcpClient:        &dns.Client{Net: "tcp", ReadTimeout: clientTimeout},
+		udpClient:        &dns.Client{Net: "udp", ReadTimeout: clientTimeout, UDPSize: udpBuffSize},
 	}
 
-	err = s.listen(address)
+	s.readUpstream()
```

Comment:
I make an attempt here so that `weave report` stands a chance of containing useful upstream infos before any recursive lookups have occurred.